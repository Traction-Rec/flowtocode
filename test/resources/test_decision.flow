<?xml version="1.0" encoding="UTF-8"?>
<Flow xmlns="http://soap.sforce.com/2006/04/metadata">
    <actionCalls>
        <name>Do_Bulk_Action</name>
        <label>Do Bulk Action</label>
        <locationX>176</locationX>
        <locationY>734</locationY>
        <actionName>ApexBulkAction</actionName>
        <actionType>apex</actionType>
        <flowTransactionModel>Automatic</flowTransactionModel>
        <nameSegment>ApexBulkAction</nameSegment>
        <versionSegment>1</versionSegment>
    </actionCalls>
    <apiVersion>60.0</apiVersion>
    <constants>
        <name>Sample_Account_Name</name>
        <dataType>String</dataType>
        <value>
            <stringValue>&quot;Sample Account Name&quot;</stringValue>
        </value>
    </constants>
    <decisions>
        <name>Account_Decision</name>
        <label>Account Decision</label>
        <locationX>396</locationX>
        <locationY>350</locationY>
        <defaultConnector>
            <targetReference>Update_Account</targetReference>
        </defaultConnector>
        <defaultConnectorLabel>Default Outcome</defaultConnectorLabel>
        <rules>
            <name>If_Account_Number_Starts_With_1</name>
            <conditionLogic>and</conditionLogic>
            <conditions>
                <leftValueReference>Account_Loop.AccountNumber</leftValueReference>
                <operator>StartsWith</operator>
                <rightValue>
                    <stringValue>1</stringValue>
                </rightValue>
            </conditions>
            <connector>
                <targetReference>Delete_Account</targetReference>
            </connector>
            <label>If Account Number starts with 1</label>
        </rules>
    </decisions>
    <environments>Default</environments>
    <interviewLabel>Test_Flow {!$Flow.CurrentDateTime}</interviewLabel>
    <label>Test_Decision_Flow</label>
    <loops>
        <name>Account_Loop</name>
        <label>Account Loop</label>
        <locationX>176</locationX>
        <locationY>242</locationY>
        <collectionReference>Accounts</collectionReference>
        <iterationOrder>Asc</iterationOrder>
        <nextValueConnector>
            <targetReference>Account_Decision</targetReference>
        </nextValueConnector>
        <noMoreValuesConnector>
            <targetReference>Do_Bulk_Action</targetReference>
        </noMoreValuesConnector>
    </loops>
    <processMetadataValues>
        <name>BuilderType</name>
        <value>
            <stringValue>LightningFlowBuilder</stringValue>
        </value>
    </processMetadataValues>
    <processMetadataValues>
        <name>CanvasMode</name>
        <value>
            <stringValue>AUTO_LAYOUT_CANVAS</stringValue>
        </value>
    </processMetadataValues>
    <processMetadataValues>
        <name>OriginBuilderType</name>
        <value>
            <stringValue>LightningFlowBuilder</stringValue>
        </value>
    </processMetadataValues>
    <processType>Flow</processType>
    <recordDeletes>
        <name>Delete_Account</name>
        <label>Delete Account</label>
        <locationX>264</locationX>
        <locationY>458</locationY>
        <connector>
            <targetReference>Account_Loop</targetReference>
        </connector>
        <inputReference>Account_Loop</inputReference>
    </recordDeletes>
    <recordLookups>
        <name>Get_Accounts</name>
        <label>Get Accounts</label>
        <locationX>176</locationX>
        <locationY>134</locationY>
        <assignNullValuesIfNoRecordsFound>false</assignNullValuesIfNoRecordsFound>
        <connector>
            <targetReference>Account_Loop</targetReference>
        </connector>
        <filterLogic>and</filterLogic>
        <filters>
            <field>Name</field>
            <operator>EqualTo</operator>
            <value>
                <elementReference>Sample_Account_Name</elementReference>
            </value>
        </filters>
        <object>Account</object>
        <outputReference>Accounts</outputReference>
        <queriedFields>Id</queriedFields>
        <queriedFields>AccountNumber</queriedFields>
    </recordLookups>
    <recordUpdates>
        <name>Update_Account</name>
        <label>Update Account</label>
        <locationX>528</locationX>
        <locationY>458</locationY>
        <connector>
            <targetReference>Account_Loop</targetReference>
        </connector>
        <inputReference>Account_Loop</inputReference>
    </recordUpdates>
    <start>
        <locationX>50</locationX>
        <locationY>0</locationY>
        <connector>
            <targetReference>Get_Accounts</targetReference>
        </connector>
    </start>
    <status>Draft</status>
    <variables>
        <name>Accounts</name>
        <dataType>SObject</dataType>
        <isCollection>true</isCollection>
        <isInput>false</isInput>
        <isOutput>false</isOutput>
        <objectType>Account</objectType>
    </variables>
</Flow>
