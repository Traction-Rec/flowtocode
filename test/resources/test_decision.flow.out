Get_Accounts
LOOP: Account_Loop
  IF: If Account Number starts with 1
    Delete_Account
  ELSE:
    Update_Account
APEX CALL: Do_Bulk_Action
